lldb/source/Plugins/Language/Swift/SwiftDictionary.cpp

```diff
@@ -45,7 +45,7 @@ DictionaryConfig::DictionaryConfig()
   // still used to name classes in the ObjC runtime.
 
   m_nativeStorageRoot_mangled =
-    ConstString("_TtCs22__RawDictionaryStorage");
+    ConstString("$ss22__RawDictionaryStorageCD");
```

Comment:
This is necessary because TypeSystemSwiftTypeRef only accepts the new mangling.

Reply:
Which part of TypeSystemSwiftTypeRef is that?

Reply:
Sorry that is incorrect. The problem is
```
swift::Type SwiftASTContext::GetSwiftType(opaque_compiler_type_t opaque_type) {
  assert(opaque_type && *reinterpret_cast<const char *>(opaque_type) != '$' &&
         "wrong type system");
```
which triggers in the VALIDATION macros when lifting the type into the SwiftASTContext type system.